### PR TITLE
implemented mars-largest-picture-app using feign client

### PR DESCRIPTION
**Pull request by vromantsev**

### CODE REVIEW

File: mars-largest-pic-feign/src/main/java/com/bobocode/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Configuration
+public class AppConfig {
+
+    @Bean
+    public RestTemplate restTemplate(final List<HttpMessageConverter<?>> converters) {
+        converters.add(new ByteArrayHttpMessageConverter());
```
Comment (luckyloker) [inline mars-largest-pic-feign/src/main/java/com/bobocode/config/AppConfig.java]: There's no need in additional converter, as produces = MediaType.IMAGE_JPEG_VALUE in GetMapping will correctly convert response.

File: mars-largest-pic-feign/src/main/java/com/bobocode/service/MarsPictureServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                .map(getPicDataFunction())
+                .max(Comparator.comparing(PicData::contentLength))
+                .map(picData -> this.restTemplate.getForObject(picData.url(), byte[].class))
+                .orElseThrow(() -> new PictureNotFoundException("No pictures found!"));
+    }
+
+    private Function<Photo, PicData> getPicDataFunction() {
+        return photo -> {
+            final String originalUrl = photo.imgSrc();
+            final HttpHeaders originalHeaders = this.restTemplate.headForHeaders(originalUrl);
```
Comment (luckyloker) [inline mars-largest-pic-feign/src/main/java/com/bobocode/service/MarsPictureServiceImpl.java:42]: Just a thought: RestTemplate can be configured with a redirect strategy for HEAD requests, so when call headForHeaders restTempalte execute an internal redirect and we can get the actual size of the picture.

**Review by luckyloker — COMMENTED** [review]
Looks great overall 👍🏻

File: mars-largest-pic-feign/src/main/java/com/bobocode/config/AppConfig.java (hunk 1 above)
Comment (vromantsev) [reply mars-largest-pic-feign/src/main/java/com/bobocode/config/AppConfig.java]: Yeah, you're right, this converter is not necessary. Moreover, rest template uses it by default on initialization stage, so I removed it.

File: mars-largest-pic-feign/src/main/java/com/bobocode/service/MarsPictureServiceImpl.java (hunk 2 above)
Comment (vromantsev) [reply mars-largest-pic-feign/src/main/java/com/bobocode/service/MarsPictureServiceImpl.java:42]: > Just a thought: RestTemplate can be configured with a redirect strategy for HEAD requests, so when call headForHeaders restTempalte execute an internal redirect and we can get the actual size of the picture.

Good point. Fixed.

**Pull request merged**